flake.nix

```diff
@@ -0,0 +1,169 @@
+                # available for use after build.
+                postInstall = ''
+                  install -D tests/bpftrace_test -t "$out/tests"
+                  install -D tests/runtime-tests.sh -t "$out/tests"
+                  install -D tests/tools-parsing-test.sh -t "$out/tests"
+                  install -D tests/memleak-tests.sh -t "$out/tests"
+                  cp -r tests/runtime "$out/tests"
+                  cp -r tests/testlibs "$out/tests"
+                  cp -r tests/testprogs "$out/tests"
+
+                  # Somewhat of a hack to trick *-test.sh into locating bpftrace binary.
+                  ln -s "$out/bin" "$out/src"
+
+                  # Nix does not seem smart enough to replace the RUNPATH our testprogs that link
+                  # to testlibs. So we need to do it manually to pass nix's later validation tests
+                  # that nothing is RPATH'd to the build directory.
+                  patchelf --set-rpath "$out/tests/testlibs" "$out/tests/testprogs/usdt_lib"
+                '';
+              };
+
+          # Create a test program that runs all tests. Ideally we'd use `ctest` but
+          # it likes to create a bunch of files and directories which the read-only
+          # nature of nix doesn't like.
+          #
+          # See: https://gitlab.kitware.com/cmake/cmake/-/issues/23231
+          runTests =
+            pkgs.writeScriptBin "run-tests.sh" ''
+              repo_root=$(git rev-parse --show-toplevel)
+              for test in "$repo_root/result/tests/"*; do
+                if [[ ! -x "$test" || -d "$test" ]]; then
+                  continue
+                fi
+
+                if [[ "$test" == *.sh ]]; then
+                  sudo "$test"
+                else
+                  "$test"
+                fi
+              done
+            '';
+        in
+        {
+          # Set formatter for `nix fmt` command
+          formatter = pkgs.nixpkgs-fmt;
+
+          # Define package set
+          packages = {
+            # Default package is bpftrace with llvm 15
+            default = mkBpftrace pkgs.llvmPackages_15;
+
+            # Support matrix of llvm versions
+            bpftrace-llvm14 = mkBpftrace pkgs.llvmPackages_14;
+            bpftrace-llvm13 = mkBpftrace pkgs.llvmPackages_13;
+            bpftrace-llvm12 = mkBpftrace pkgs.llvmPackages_12;
+            bpftrace-llvm11 = mkBpftrace pkgs.llvmPackages_11;
+            bpftrace-llvm10 = mkBpftrace pkgs.llvmPackages_10;
+          };
+
+          # Define the development shell you get when you run `nix develop`
+          devShells.default =
+            with import nixpkgs { inherit system; };
+            pkgs.mkShell rec {
+              packages = with pkgs; [
+                runTests
+                git
+                self.packages.x86_64-linux.default.nativeBuildInputs
+                self.packages.x86_64-linux.default.buildInputs
```

Comment:
If these stay, `s/x86_64-linux/"${system}"/`

Reply:
`s/x86_64-linux/${system}/` should be fine (the quotes aren't necessary)

Reply:
Ack. Good catch.